tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs

```diff
@@ -190,5 +190,42 @@ public async Task Scenario3(string inputJson)
                 DirectoryHelper.DeleteGitDirectory(testFolder);
             }
         }
+
+        [EnvironmentConditionalSkipTheory]
+        [InlineData(
+        @"{
+              ""AssetsRepo"": ""Azure/azure-sdk-assets-integration"",
+              ""AssetsRepoPrefixPath"": ""pull/scenarios"",
+              ""AssetsRepoId"": """",
+              ""TagPrefix"": ""main"",
+              ""Tag"": ""language/tables_bb2223""
+        }")]
+        [Trait("Category", "Integration")]
+        public async Task RestoreUpdatesContext(string inputJson)
+        {
+            var folderStructure = new string[]
+            {
+                GitStoretests.AssetsJson
+            };
+
+            Assets assets = System.Text.Json.JsonSerializer.Deserialize<Assets>(inputJson);
+            var testFolder = TestHelpers.DescribeTestFolder(assets, folderStructure);
+            var pathToAssets = Path.Join(testFolder, "assets.json");
+            var currentDirectory = Environment.CurrentDirectory;
+            var recordingHandler = new RecordingHandler(currentDirectory, _defaultStore);
+            
+            try
+            {
+                await recordingHandler.Restore(pathToAssets);
+
+                var result = (await _defaultStore.ParseConfigurationFile(pathToAssets)).AssetsRepoLocation;
+
+                Assert.Equal(result, recordingHandler.ContextDirectory);
```

Comment:
We should also assert the file count and file contents, you should be able to copy those directly from Scenario3.

Reply:
Agree.